### ISSUE TITLE
Slopes' right side culled at certain relative position (not angle)

### ISSUE THREAD

**Issue by No3371** [quat1024/templates-mod#6]
https://streamable.com/z9vg9m

- More Culling already turned off for Templates 2.
- MC 1.20.1
- Sodium 0.5.3
- Indium 1.0.27
- Iris 1.6.10
- I'm kinda heavily modded though, if it's not reproduce-able for you I may scan my mods to check conflicting mods.

BTW I have been looking for similar mods for 1.20.1, with keywords like "fake", "frame"... no luck, unintentionally just found Templates 2 and it seems to be the best option for 1.20.1 now, good work.

**Comment by tanyan-tanyan** [comment]
I have all of these mods except Iris and I have the same issue. I'm also heavily modded unfortunately :(

![javaw_puatjtEohe](https://github.com/quat1024/templates-mod/assets/68793071/ac988280-9226-49ba-9e60-816af5a21a09)

**Comment by quat1024** [comment]
Hmm, think i might know what's causing this now and just reproduced it on the blanketcon test server. Some combination of these things:

* just guessing, but there might be some mod that culls the backfaces of blocks? (maybe Sodium?)
* I don't think I set the cullshape correctly for my Templates blocks
* I also don't think I set the cull*faces* correctly on the Slope model. The slope model is a bit of a mess tbh

Basically when you get far away from the slope, this mod thinks that face that's pointing towards you is actually on the back of the block and hides it, which is my bad because I'm feeding the game bad data

**Comment by quat1024** [comment]
Ok I think the cullshape is "fine" (I was thinking about a different problem with MoreCulling, which doesn't respect the block's render layer and culls stuff behind it anyway)

Maybe the left and right sides of the slope simply had the wrong cullface + I was also permuting some data that didn't need to be permuted around when rotating the model.

Fixing that sent me down a rabbit hole of fixing AO on the slopes 🤔 💭

**Comment by quat1024** [comment]
Couldn't get AO fixed (that will be its own adventure and probably require rearchitecting some bits of the mod honestly) but this should be fixed in https://modrinth.com/mod/templates-2/version/2.5.1-fabric-1.21.1

Needs a 1.20 backport though ://////// ugh
